packages/rich-text/src/component/use-anchor.js

```diff
@@ -174,7 +148,35 @@ export function useAnchor( { editableContentElement, settings = {} } ) {
 			editableContentElement.removeEventListener( 'focusin', attach );
 			editableContentElement.removeEventListener( 'focusout', detach );
 		};
-	}, [ editableContentElement, tagName, className ] );
+	}, [ editableContentElement, tagName, className, callback ] );
+
+	return {
+		anchor,
+		update: callback,
+	};
+}
+
+/**
+ * This hook, to be used in a format type's Edit component, returns the active
+ * element that is formatted, or a virtual element for the selection range if
+ * no format is active. The returned value is meant to be used for positioning
+ * UI, e.g. by passing it to the `Popover` component via the `anchor` prop.
+ *
+ * @param {Object}           $1                        Named parameters.
+ * @param {HTMLElement|null} $1.editableContentElement The element containing
+ *                                                     the editable content.
+ * @param {WPFormat=}        $1.settings               The format type's settings.
+ * @return {Element|VirtualAnchorElement|undefined|null} The active element or selection range.
+ */
+export function useAnchor( { editableContentElement, settings = {} } ) {
+	const { anchor } = useAnchorBase( { editableContentElement, settings } );
 
 	return anchor;
 }
+
+export function useAnchorWithUpdate( {
+	editableContentElement,
+	settings = {},
+} ) {
+	return useAnchorBase( { editableContentElement, settings } );
+}
```

Comment:
This is our new private version of `useAnchor`. The only difference is that it exposes the full API of the "base" hook. The important part is that we expose the `update` method as part of the returned object.

We could selectively only expose `anchor` and `callback` here - i'm flexible about that.